sql/core/src/test/scala/org/apache/spark/sql/StatisticsCollectionSuite.scala

```diff
@@ -204,6 +204,24 @@ class StatisticsCollectionSuite extends StatisticsCollectionTestBase with Shared
     }
   }
 
+  test("SPARK-25028: column stats collection for null partitioning columns") {
+    val table = "analyze_partition_with_null"
+    withTempDir { dir =>
+      withTable(table) {
+        sql(s"""
+             |CREATE TABLE $table (name string, value string)
+             |USING PARQUET
+             |PARTITIONED BY (name)
+             |LOCATION '${dir.toURI}'""".stripMargin)
+        val df = Seq(("a", null), ("b", null)).toDF("value", "name")
```

Comment:
super nit: better to add a non-null partition value, e.g., `val df = Seq(("a", null), ("b", null), ("c", "1")).toDF("value", "name")`? btw, why is this a reverse column order (not "name", "value", but "value", "name")?

Reply:
I don't think it is needed to add another partition value, as the problem here is with `null` throwing an NPE and the test shows that no NPE is thrown. But if you think it is necessary I can add it.

The reverse column order is the way spark works when inserting data into a partitioned table. The partitioning columns are specified at the end, after the non-partitioning ones.

Reply:
when creating the table, we can put partition column at the end, to avoid this confusion.

Reply:
ok, will do, thanks.